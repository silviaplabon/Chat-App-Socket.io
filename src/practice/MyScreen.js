import React, {useState} from 'react';
import {StyleSheet, Text, View,Button} from 'react-native';

const MyScreen=()=>{
    const [status,setStatus]=useState('');
    return (
      <View  style={styles.container}>
           <Text testID="myText">{status}</Text>
           <Button testID="myButton" onPress={()=>setStatus('button pressed')} title="pressMe"></Button>
      </View>
    );
  }


const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    flex: 1,
    justifyContent: 'center',
  },
  instructions: {
    color: '#333333',
    marginBottom: 5,
    textAlign: 'center',
  },
  welcome: {
    fontSize: 20,
    margin: 10,
    textAlign: 'center',
  },
});

export default MyScreen;