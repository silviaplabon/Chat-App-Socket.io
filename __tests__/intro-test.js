import React from 'react';
import renderer from 'react-test-renderer';
import MyScreen from '../src/practice/MyScreen';

const tree = renderer.create(< MyScreen/>)


test('button press',()=>{
    const button=tree.root.findByProps({testID:'myButton'}).props;
    renderer.act(()=>button.onPress());
    const text=tree.root.findByProps({testID:'myText'}).props;
    expect(text.children).toEqual('button pressed');
    expect(tree).toMatchSnapshot()
})