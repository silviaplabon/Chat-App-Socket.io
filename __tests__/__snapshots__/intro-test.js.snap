// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button press 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    testID="myText"
  >
    button pressed
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        pressMe
      </Text>
    </View>
  </View>
</View>
`;
